File: day_03/part02.py
#!/usr/bin/env python3

# advent of code 2017: day 03, part 02

INPUT = 325489

size = 21
g = [[0 for i in range(size)] for j in range(size)]
g[size//2][size//2] = 1
sq=3
i=size//2+1
j=size//2+1
moves = [(-1,0),(0,-1),(1,0),(0,1)]
done = False
while not done:
    for side in range(4):
        for _ in range(sq-1):
            x,y = moves[side]
            i += x
            j += y            
            n = 0
            for x in range(-1,2):
                for y in range(-1,2):
                    n += g[i+x][j+y]
            g[i][j] = n
            if n >= INPUT and done == False:
                print("result", n)
                done = True
    sq += 2
    i += 1
    j += 1

File: day_02/part02.go
/* advent of code 2017: day 02, part 02 */
package main

import (
	"bufio"
	"fmt"
	"io"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {
	fp, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer fp.Close()
	rd := bufio.NewReader(fp)
	res := 0
	for {
		str, err := rd.ReadString('\n')
		if err != nil {
			if err == io.EOF {
				break
			}
			log.Fatal(err)
		}
		str = strings.Trim(str, " \t\n\r\v\f")
		split := strings.Split(str, "\t")
		array := make([]int, len(split))
		for i, e := range split {
			val, _ := strconv.Atoi(e)
			array[i] = val
		}
		done := false
		for i := 0; i+1 < len(array) && !done; i++ {
			for j := i + 1; j < len(array) && !done; j++ {
				var big, small int
				big, small = array[i], array[j]
				if small > big {
					big, small = array[j], array[i]
				}
				if big%small == 0 {
					done = true
					res += big / small
				}
			}
		}
	}
	fmt.Println(res)
}
